src/main.py

```diff
@@ -171,13 +173,25 @@ def _run_testing(env: BaseEnv, approach: BaseApproach) -> Metrics:
     metrics["num_total"] = len(test_tasks)
     metrics["avg_suc_time"] = (total_suc_time /
                                num_solved if num_solved > 0 else float("inf"))
+    total_skeletons_optimized = approach.metrics[
+        "total_num_skeletons_optimized"]
+    metrics["avg_skeletons_optimized"] = (
+        total_skeletons_optimized /
+        num_found_policy if num_found_policy > 0 else float("inf"))
+    metrics["min_skeletons_optimized"] = approach.metrics[
+        "min_num_skeletons_optimized"]
+    metrics["max_skeletons_optimized"] = approach.metrics[
+        "max_num_skeletons_optimized"]
     total_num_nodes_expanded = approach.metrics["total_num_nodes_expanded"]
     metrics["avg_nodes_expanded"] = (total_num_nodes_expanded /
                                      num_found_policy
                                      if num_found_policy > 0 else float("inf"))
     total_plan_length = approach.metrics["total_plan_length"]
     metrics["avg_plan_length"] = (total_plan_length / num_found_policy
                                   if num_found_policy > 0 else float("inf"))
+    metrics["avg_execution_failures"] = (
+        total_num_execution_failures /
+        num_found_policy if num_found_policy > 0 else float("inf"))
```

Comment:
note: important to be careful here with the denominator, esp. whether it should be `num_found_policy` or `num_solved` or something else entirely. depends what you're thinking of when you think about what this average means exactly.

Reply:
good note, thanks!